### DISCUSSION TITLE
Defensebot's inheritteam doesn't work in outpost deathmatch (plus several more issues... again)

### DISCUSSION THREAD

**Discussion by girlyguppy** [FakeFishGames/Barotrauma#14838] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

defensebot doesn't attack deathmatchteam2 if made by a person on deathmatchteam1. or vice versa. (I'm assuming inheritteam is inheriting it from the submarine's team, not from the player who fabricated it? on outposts, there is only 1 submarine team)

additionally, the defense bot's idcard can't be edited in its properties in subeditor.
or the ammo.  
and it's hideinmenus="true" so it has to be spawnitem'd

and another issue, in sub v sub mode, while it does work against enemy team, you can simply grab it to stop it from shooting you (too close to your body), and also remove the ammo or idcard.

in general just please support defensebots in subeditor and custom games🙏

### Reproduction steps

1. be in outpost deathmatch
2. be either a separatist or coalition
3. craft a defensebot (requires you to use 'givetalent Robotics' command because medals, xp, and talents isn't allowed in outpost deathmatch for some reason)
4. watch as it doesn't shoot at the opposite team

### Bug prevalence

Happens every time I play

### Single player or multiplayer?

Multiplayer hosted from the in-game menu (= using a listen server)

### Version

v1.6.4.0 (unstable)

### Which operating system did you encounter this bug on?

Windows

**Comment by AziKarnage** [comment]
Vampire men are awhtgngjfmghfndf

**Reply by AziKarnage** [reply]
Girlypuppy is a hater and doesn't understand the real power of a suckulent immortal Twink with super strength

**Comment by itchyOwl** [comment]
Fixed the team and made defensebot placeable in the sub editor (FakeFishGames/Barotrauma-development#5902). Also fixed being able to grab hostile pets.

Defensebot's id card or ammo, or anything in the inventory can't be edited in the editor, because the item we place is just a spawner item. The character doesn't exist at this point, so the inventory can't be edited either. Can't think of an easy way to change this. However, you can use events and spawn points to give defensebots access to places. For an example of spawning a defensebot via events, see the "malfunctioningdefensebot" event. The tags defined in the spawn point should be added to the id card tags of any spawned character. Let us know if it doesn't work.

**Reply by girlyguppy** [reply]
Petraptor also is currently having the same issue (not attacking enemy team)

Using events is not something I really know how to do, so I can't test that myself.

Is it possible for the defensebot to inherit the inventory of the defensebotspawner item? 
Possibly a way to find a deactivated one (item), and then interact to activate (spawn) and bring it to your team (rather than it instantly spawning on round start)?

**Comment by Regalis11** [comment]
Fixed in our private repo now, fix coming up in the next unstable build and the upcoming release.
